### ISSUE TITLE
Translations of the Guide

### ISSUE THREAD

**Issue by xmalinov** [braydie/HowToBeAProgrammer#45]
Is that a good idea to translate it? IMHO, it would be interesting in Russian (or Chinese)

**Comment by braydie** [comment]
This is a great idea @xmalinov! There's been a reasonable amount of support to move to gitbook so that we have an improved reading/writing experience, and fortunately [gitbook supports multiple languages](https://help.gitbook.com/format/languages.html) so we might be on to a winner with that one :smile:

**Comment by UbuntuLover** [comment]
是啊 如果有中文版本的会好很多的

**Comment by rccoder** [comment]
translate into Chinese？
I think I can help.

**Comment by xmalinov** [comment]
I have a small tip for better translating process: to adopt the Semantic Versioning for our purposes and use it for the Guide. It can make sync of the translation with the original document a bit easy. The translated version will be refer to a specific version.
E.g.:
MAJOR.MINOR.PATCH, increment the:
MAJOR version when you make the gitbook release of newer edition,
MINOR version when you add a chapter or a whole section
PATCH version when you make typo fixes.

**Comment by adhamenaya** [comment]
Great idea.. I can translate it to Arabic as well

**Comment by ImDineshSaini** [comment]
I can translate in Hindi India

**Comment by henriquesosa** [comment]
I can translate to Portuguese.

**Comment by huanzhang** [comment]
I can translate it to Chinese as well. @rccoder :)

**Comment by RobertLRead** [comment]
I believe it has already been translated into Korea and Belarussian. I can't seem to find the translations however.  I would of course be honored for anyone to undertake a translation.

**Comment by ujuc** [comment]
@RobertLRead 
This Koean version. to old...
https://wiki.kldp.org/wiki.php/HowToBeAProgrammer

**Comment by bigla** [comment]
I could start translating it to german.

**Comment by UbuntuLover** [comment]
@huanzhang  looking forward your Chinese version!

**Comment by Minipada** [comment]
I can start the french one

**Comment by ImDineshSaini** [comment]
Please let me know if I can start in Hindi

**Comment by henriquesosa** [comment]
i already get started to translate it into pt-br

**Comment by braydie** [comment]
This is all very exciting -- with the conversion to gitbook I think we'll be in a better position to provide the guide in multiple languages ([gitbook multiple languages](https://help.gitbook.com/format/languages.html)). I'll look to get the [gitbook conversion](https://github.com/braydie/HowToBeAProgrammer/pull/44) merged in, and then restructure for multiple languages

**Comment by braydie** [comment]
I've merged the changes in to move us over to a gitbook-compatible repository so we are now in a position to start accepting pull requests for translations :+1:

**Comment by huanzhang** [comment]
:+1:

**Comment by kuangchanglang** [comment]
@rccoder just do it

**Comment by ahangchen** [comment]
I‘m translating it into Chinese too...
https://github.com/ahangchen/How-to-Be-A-Programmer-CN

**Comment by huanzhang** [comment]
@ahangchen :+1:

**Comment by braydie** [comment]
@ahangchen amazing! :+1:

**Comment by adhamenaya** [comment]
I's translating the guide into Arabic as well...
[https://github.com/adhamenaya/HowToBeAProgrammer-AR ](url)

**Comment by ahangchen** [comment]
Chinese version done
https://github.com/ahangchen/How-to-Be-A-Programmer-CN

**Comment by braydie** [comment]
Wow @ahangchen this is awesome! Would you be happy for me to lift what you've done into this repository and set it up so that it is a multilingual gitbook?

@adhamenaya your link didn't work for me.. has the repo been deleted?

**Comment by huanzhang** [comment]
@ahangchen :+1:

**Comment by ahangchen** [comment]
@braydie Of course, I'm glad to do that.

**Comment by braydie** [comment]
Thank you @ahangchen, it's now merged and available via gitbook :smile:

**Comment by ahangchen** [comment]
@braydie  :+1:

**Comment by jewirth** [comment]
@xmalinov initially asked whether it's a good idea to add translations. This issue is without any contradiction yet. So I'd like to add another point of view here just for the sake of diversity of opinions:

Personally, I couldn't image being a programmer without understanding written English.
Hence, does it make sense to learn how to program without understanding English at all?

I see severe drawbacks for those who try:
- learning: no doubt, most programming resources are English. Talks, Conferences, Books, Blogs...
- working: parameters, syntax-statements, comments in sources, error messages are mostly English
- creating: a programmer who creates source code using non-English variable names, comments and documentation creates products which only a minority of programmer's can maintenance

Well, I don't think translating is a bad idea. But it's not good either :-)

**Idea: How's about adding a strong advice to learn English first - given the reasons above - into the translated versions?**

**Comment by haydarai** [comment]
I'll do a translation for Indonesia version.

**Comment by braydie** [comment]
@jewirth that's interesting - I think initially there was no reason to not offer the guide in as many languages as possible. I agree that English is essential to be a professional programmer today, and I think there is enough to discuss on that topic to warrant it's own section (your thoughts @RobertLRead?). 

This guide is quite conceptual - it doesn't really relate directly to a particular area of development, nor does it offer any code snippets. With that in mind, I think offering these concepts in as many languages as possible is a good thing - otherwise, you're relying on somebody having an excellent understanding of English before being able to embark on the journey towards being a programmer. Do you think somebody wouldn't be able to gain a basic understanding of  the concept of debugging before learning English? I agree that actually working with code would be difficult (probably impossible), but *conceptually* somebody could understand it before actually using it in earnest. 

Happy to keep this dialogue open, though - such an interesting idea. Thanks for providing another angle on this :smile:

**Comment by Nimdis** [comment]
@xmalinov I would like to participate in the translation to Russian

**Comment by Nimdis** [comment]
@jewirth I'am familiar with many people in Russia that understand english very bad despite this they are very good programmers. And after all this would be useful for children or students.

**Comment by braydie** [comment]
@xmalinov re: semantic versioning - 100% agree. Suppose it makes to sense to version each page as this will allow for easy comparison across languages to see if a page is out of date

**Comment by ahangchen** [comment]
I have a suggestion. 

Checking for the pull requests without understanding corresponding language is not a good idea.  

Fixes for other languages version should first be push to that languages branch. After checked up by the origin translator,  then it can be merged into this repository. 

For example, adding a declaration like this:
> If you want to fix the Chinese version, please open a pull request here: https://github.com/ahangchen/HowToBeAProgrammer

**Comment by braydie** [comment]
@ahangchen great idea getting a second set of eyes over each translation

**Comment by braydie** [comment]
In pull request #79 we've tried to follow the new way of handling translations - via a pull request on the original translation repository. It looks like it's probably not the most logical way of working, so I'm proposing the following:

**If you provide the initial translation of the guide into another language, you will be made a contributor on this project and given the responsibility of reviewing future pull requests to 'your' translation**

This will give us the best of both worlds - a single place to contribute, and a chance to review translations by more than one person.

**Comment by SZain** [comment]
I agree with @braydie from your last comment. So shall I begin translating in Urdu then ? (I personally struggle to understand some terms or meaning in English. I usually translate or get the meaning of words/terms from english to my language in order to understand. I think this is a great idea.

**Comment by braydie** [comment]
@SZain I'd love to see it translated into Urdu - that would be awesome! If you feel you might struggle with translating 100% accurately, just submit the pull request and hopefully we can get some collaborators on it to help :smile:

**Comment by braydie** [comment]
Hit enter accidently when making the change - but I've updated the Contributions page to reflect the new guidance on translations - https://github.com/braydie/HowToBeAProgrammer/commit/054e0023a6724445d582a2ca43eebadc914adfd8

**Comment by ali4ever4** [comment]
I am also interested in contributing to translate into Arabic @adhamenaya please let me know how I can support your work in translation to Arabic and how you are following version updates in the original guid ?

**Comment by jimmysoda** [comment]
I started translating to Spanish:
https://github.com/jimmysoda/HowToBeAProgrammer-ES

**Comment by rraulg5** [comment]
I can help with translation to Spanish

**Comment by yammesicka** [comment]
I will start translating it to Hebrew soon.

**Comment by gorohoroh** [comment]
Is it true that no one has started work on a Russian translation so far?

**Comment by Robot72** [comment]
I decided to try it. I began to translate into Russian.

**Comment by girol** [comment]
The project is still accepting translations?

**Comment by braydie** [comment]
Hi @girol we are indeed! At present we've got English, Chinese and Japanese.

**Comment by girol** [comment]
@braydie fantastic! How do I contribute? I'm thinking about  pt-br translation. The other translation project seems to be abandoned. Anyone correct me if I'm wrong. 🐭

**Comment by braydie** [comment]
Best way to get started would be to fork the repository and in your own copy create a new folder named 'pt-br' and take a copy of the chapters and sections from the English version to translate from. In order for gitbook to pick up the new translation you'll also need to update the [langs file at the root of the repository](https://github.com/braydie/HowToBeAProgrammer/blob/master/LANGS.md). 

Once your translation is complete you can submit it via a pull request!

**Comment by braydie** [comment]
Also once we've had a language added we typically add the person who provided the translation as a collaborator on the project so that we've got someone who can translate any additional changes in the future

**Comment by Luc45** [comment]
Still need PT-BR translation?

**Comment by ohahohah** [comment]
I started to translate this into Korean. If anyone is interested, here is https://github.com/ohahohah/HowToBeAProgrammer/tree/master/kr

**Comment by braydie** [comment]
That's awesome @ohahohah - when you're ready let me know and we can get it integrated into the main project! Thanks for your contribution :+1:
